content/preferences-settings/shortcuts.md

```diff
@@ -5,133 +5,204 @@ weight: 120
 
-## views
+Such point-and-click type _actions_ are normally defined as the application of an _effect_ to an _element_ of a _widget_, where these terms are defined as follows:
 
-A single section is provided for each darktable view. Shortcut actions are only applicable to the selected view.
+widget
+: Each visible part of the user interface is known as a _widget_. For example the darktable application window is a widget, containing side panel widgets, each of which contains module widgets, each of which contains button, slider and dropdown list widgets etc... When assigning a shortcut to an action, you must first decide which widget it is to be applied to.
 
-## processing modules
+element
+: An _element_ is the part of a UI widget that is affected by your shortcut. For example, for a slider that has a color picker, you can make a shortcut activate the color picker _button_ element or change the _value_ element of the slider. For a row of tabs (the row is a single widget) you can select which tab element to activate.
 
-Shortcut actions in this category apply to [processing modules](../module-reference/processing-modules/_index.md) in the darkroom view. A section is provided for each processing module.
+effect
+: A shortcut can sometimes have multiple possible _effects_ on a given _element_. For example, a button can be activated as if it was pressed with a plain mouse-click or as if it was pressed with Ctrl+click. A slider's value can be edited, increased/decreased or reset.
 
-In addition, a separate "blending" section allows you to control the [masking and blending](../darkroom/masking-and-blending/_index.md) options for the currently active module.
+# assigning shortcuts to actions
 
-### common shortcuts
+There are two primary methods of assigning a shortcut to an action.
 
-Every processing module provides the following shortcut actions by default:
+## visual shortcut mapping
 
-enable module
-: Enable or disable the module, regardless of whether it is currently visible.
+Click on the ![visual mapping button](./shortcuts/visual-mapping-button.png#icon) in the [top panel](../overview/user-interface/top-panel.md) of any darktable view to enter visual shortcut mapping mode. 
 
-show module
-: Expand or collapse the module. If the module is not currently displayed on the screen, darktable will switch to an appropriate module group before displaying it.
+The mouse cursor will change as you hover over UI widgets, to indicate whether or not a mapping can be created:
 
-focus module
-: Cause the module to receive or lose focus. This can activate on-screen guides or controls for that module (for example you can focus the crop module to show the crop guides and unfocus it to commit settings).
+- A cross ![cross icon](./shortcuts/cross-icon.png#icon) indicates that there is no mappable widget under the cursor. Clicking the left mouse button will call up the shortcut mapping screen (where you can configure shortcuts for non-visual actions).
```

Comment:
where you can _also_ configure shortcuts for non-visual actions.

Maybe if you reverse the order of the different cursors and treat this one last, you can mention left-click under "spiral" to edit settings for the widget and then have "cross" at the end say if no mappable widget under cursor, the dialog is called up with a selection depending on the part of the screen that was clicked on (global, view, libs or mods).